### ISSUE TITLE
Java null pointer exception when executing .sql files

### ISSUE THREAD

**Issue by Simnbf** [IBM/db2forzosdeveloperextension-about#78]
## Development environment where the bug occurred

- Db2 Developer Extension version:
- Editor platform
  - [X] Visual Studio Code
  - [ ] Eclipse Theia
- Editor platform version: 1.66.0
- Operating system on which your editor runs (for example, Windows 10 2004 or MacOS Catalina 10.15.7): Windows 10
- Java Version (Run `java -version` and paste the details here):
```
$ java -version
openjdk version "11.0.9" 2020-10-20 LTS
OpenJDK Runtime Environment Zulu11.43+22-SA (build 11.0.9+11-LTS)
OpenJDK 64-Bit Server VM Zulu11.43+22-SA (build 11.0.9+11-LTS, mixed mode)
```
- Db2 for z/OS version (including function level for Db2 12): Db2 V12R1M501
- [Log files](https://ibm.github.io/db2forzosdeveloperextension-about/docs/troubleshooting/opening-an-issue-and-providing-logs) attached?: Yes

## Problem description

Detailed steps for reproducing the problem:
[server_2022-03-31T131648.log](https://github.com/IBM/db2forzosdeveloperextension-about/files/8387202/server_2022-03-31T131648.log)

1. First step

User updated VS code from 1.65.2 to 1.66 this morning, right click .sql file and 'Run All' results in a java exception, see below

### Expected behavior

SQL should execute and return result set, worked fine under 1.65.2 

 
[server_2022-03-31T131648.log](https://github.com/IBM/db2forzosdeveloperextension-about/files/8387207/server_2022-03-31T131648.log)

```
Db2SqlService:initialize: Initializing Db2SqlService.

[Info  - 12:54:12 PM] Db2SqlService:loadConfiguration: Configurations loaded {"port":"55555","maxHeapSize":"256M","dependencies":""}.

[Info  - 12:54:12 PM] EnvironmentManager:getPlatform: Found platform config_win for win32.

Db2SqlServiceClient:createServerConnection: Attempting to connect to Db2SqlService server at port 55555; attempt 1.

EnvironmentManager:getJavaPath: Found db2forzosdeveloperextension.java.home user setting in Preferences with value: C:\apps\zulu\jdk\zulu11.43.22-sa-jdk11.0.9-win_x64\. Will use that to locate a valid Java SDK for language servers.

Db2SqlServiceClient:createServerProcess: Starting Db2 SQL Service process with runtime C:\apps\zulu\jdk\zulu11.43.22-sa-jdk11.0.9-win_x64\bin\java ["-Dfile.encoding=UTF-8","-cp","c:/Users/G01156572/.vscode/extensions/ibm.db2forzosdeveloperextension-1.3.4/dist/dss-dist-0.0.1-SNAPSHOT.jar;;c:\\Users\\G01156572\\.vscode\\extensions\\ibm.db2forzosdeveloperextension-1.3.4\\lib\\*","com.ibm.db2.server.DssLauncher",55555,"off","c:\\Users\\G01156572\\AppData\\Roaming\\Code\\logs\\20220331T074237\\exthost1\\IBM.db2forzosdeveloperextension","-Xmx256M"]

[Info  - 12:54:14 PM] Db2SqlServiceClient:createServerProcess: Db2 SQL Service server process started on pid: 26932

[Info  - 12:54:16 PM] Db2SqlServiceClient:process:stdout: The server is running on port 55555

 

[Info  - 12:54:16 PM] Db2SqlServiceClient:process: Db2SqlService server took 3474ms to start up.

[Info  - 12:54:16 PM] Db2SqlServiceClient:createServerConnection: Server connection established after 1 attempts

[Info  - 12:54:16 PM] Db2SqlService:createClient: DSS client ready, sending connection key db2forzosdeveloperextension.

Db2SqlServiceClient:sendNotification: Send notification received (server/setConnectionKey), waiting on connection.

Db2SqlServiceClient:sendNotification: Connection confirmed, sending notification (server/setConnectionKey).

TextDocumentHandler:validateTextDocumentHandler: Validating text document [file:///c%3A/Users/G01156572/Desktop/VSC/test.sql]()

TextDocumentHandler:validateTextDocumentHandler: Retrieved document settings {"db2sqlservice":{"connectionKeepAlive":300,"debugger":{"sessionManagerPort":"4555"},"dependencies":"","maxHeapSize":"256M","maxNumberOfRowsReturned":-1,"port":"55555"},"debug":{"logLevel":"off"},"java":{"home":"C:\\apps\\zulu\\jdk\\zulu11.43.22-sa-jdk11.0.9-win_x64\\"},"maxNumberOfProblems":100,"trace":{"server":"off"},"exportOptions":{"delimiter":",","quote":"\"","escapedQuote":"\"\"","trimLeadingBlanks":false,"trimTrailingBlanks":false}}

TextDocumentManager:getSqlSettings: Received resource [file:///c%3A/Users/G01156572/Desktop/VSC/test.sql]()

TextDocumentManager:getSqlSettings: Retrieving sql settings from [file:///c%3A/Users/G01156572/Desktop/VSC/]() for test.sql

Db2SqlServiceClient:sendRequest: Send request received (server/parse), waiting on connection.

Db2SqlServiceClient:sendRequest: Connection confirmed, sending request (server/parse).

[Info  - 12:54:38 PM] Db2SqlServiceClient:process:stdout: {"requestType":"parse","properties":{"callerId":"ibm.db2forzosdeveloperextension"}}

 

TextDocumentHandler:executeQueryHandler: Execute query settings: {"connectionName":"3667aa16-5ed9-4111-be6c-f74f3f8934b6","query":"SELECT * FROM WBYTN.TSQBRGM;","statements":[{"position":{"startOffset":0,"endOffset":26,"startLine":1,"endLine":1,"startCharacter":1,"endCharacter":27},"type":3,"variables":[]}],"config":{"connectionOptions":"UseNewConnection","onSuccessOptions":"CommitOnEachStmt","onErrorOptions":"OnErrStop","terminator":";","currentSchema":"","currentPath":"","useUpperCaseOption":true,"maxNumberOfRowsReturned":-1},"modifiedVariables":[],"resultFilePath":"c:\\Users\\G01156572\\AppData\\Roaming\\Code\\User\\globalStorage\\ibm.db2forzosdeveloperextension\\sql_execution\\test_sql\\2022-03-31T072438.json"}

Db2SqlServiceClient:sendRequest: Send request received (server/getBackEndExecResult), waiting on connection.

Db2SqlServiceClient:sendRequest: Connection confirmed, sending request (server/getBackEndExecResult).

[Error - 12:54:39 PM] Db2SqlServiceClient:process:stderr: Mar 31, 2022 12:54:39 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError

SEVERE: Internal error: java.lang.NullPointerException

java.util.concurrent.CompletionException: java.lang.NullPointerException

    at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)

    at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)

    at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1702)

    at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)

    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)

    at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)

    at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)

    at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)

    at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

Caused by: java.lang.NullPointerException

    at com.ibm.db2.server.DssServerImpl.lambda$getBackEndExecResult$8(DssServerImpl.java:225)

    at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)

    ... 6 more
```

**Comment by Simnbf** [comment]
Attaching some additional logs
[additional-logs.txt](https://github.com/IBM/db2forzosdeveloperextension-about/files/8390853/additional-logs.txt)

**Comment by scotdb** [comment]
Experiencing the same here, also following a VS Code upgrade to 1.66.0 last night.

Deleted / reinstalled the extension and was able to run one SQL statement before it started happening again - very strange.

**Comment by katsoohoo** [comment]
Hi @Simnbf,

Thanks for reporting this issue. We are able to recreate on our side with VS Code 1.66 and will be looking into the issue.

**Comment by katsoohoo** [comment]
Hi @Simnbf and @scotdb,

We've released a fix for this in v1.3.5. Please let us know if it works for you.

**Comment by scotdb** [comment]
Works for me.  Thanks for the quick turnaround.
